tools/stress-cluster/services/Stress.Watcher/src/PodEventHandler.cs

```diff
@@ -215,5 +228,78 @@ public bool ShouldStartPodChaos(WatchEventType type, V1Pod pod)
 
             return true;
         }
+
+        public async Task DeleteResources(V1Pod pod)
+        {
+            if (!ShouldDeleteResources(pod))
+            {
+                Logger.Debug($"Skipping pod deletion for pod {pod.Name()}.");
+                return;
+            }
+
+            var rgName = GetResourceGroupName(pod);
+
+            if (rgName == "")
+            {
+                return;
+            }
+
+            Subscription subscription = ARMClient.DefaultSubscription;
+
+            ResourceGroup resourceGroup;
+            try {
+                resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
+            } catch (Exception e){
+                Logger.Error($"Failed to get resource group {rgName} using subsription id {subscription.Id}");
+                Logger.Error($"Throwing exception: {e}");
+                return;
```

Comment:
I would just re-throw and let our top level handler print out the full failure.

```suggestion
                throw e;
```

Reply:
Wouldn't this just terminate the whole stress test watcher?